client/app/bundles/KudosApp/components/Kudo.jsx

```diff
@@ -1,127 +1,175 @@
-import React, { PropTypes } from 'react';
-import { map } from 'lodash';
-import moment from 'moment';
-import { grey400, lightBlue400 } from 'material-ui/styles/colors';
-import FloatingActionButton from 'material-ui/FloatingActionButton';
-import ThumbUp from 'material-ui/svg-icons/action/thumb-up';
+import React, { PropTypes } from 'react'
+import { map, chunk, isEmpty } from 'lodash'
+import moment from 'moment'
+import { Tooltip } from 'material-ui'
+
+const UserAvatar = ({ user }) => (
+  <Tooltip
+    placement="top"
+    className="kudo__tooltip"
+    title={ <div className="kudo__tooltip-text">{ user.name }</div> }>
+    <img src={ user.avatar } alt={ user.name } className="avatar" />
+  </Tooltip>
+)
 
 export default class Kudo extends React.Component {
-  constructor(props, context) {
-    super(props, context);
+  constructor(props) {
+    super(props);
     this.state = {
-      likeAction: this.props.likeKudo(this.props.id),
-      timestamp: this.formatTimestamp(this.props.kudo.given_at),
-      thumbColor: grey400,
-      likeText: this.formatLikeText((this.props.kudo.likes ? this.props.kudo.likes.length : 0)),
       body: this.props.kudo.body,
-      editing: false
+      editing: false,
     };
+  }
 
-    if (this.likedBySelf(this.props.kudo.likes, this.props.giverId)) {
-      this.state.thumbColor = lightBlue400;
-      this.state.likeAction = this.props.unlikeKudo(this.props.id);
-    }
+  makeEditable = () => {
+    this.setState({ editing: true })
   }
 
-  formatTimestamp = (t) => {
-    let ts = moment(t);
-    return `At ${ts.format('h:mm a')} on ${ts.format('MMM D, YYYY')}`;
+  setMessage = (e) => {
+    this.setState({ body: e.target.value })
   }
 
-  formatLikeText = (numLikes) => {
-    if (numLikes === 0) {
-      return "";
-    }
-    return `${numLikes} ${numLikes === 1 ? 'person likes': 'people like'} this`;
+  update = () => {
+    this.setState({ editing: false })
+    this.props.updateKudo(this.props.kudo.id, this.state.body)
   }
 
-  likedBySelf = (likes, giverId) => {
-    return likes.some(like => like.giver_id === this.props.giverId)
+  formattedHeaderText() {
+    const recipients = map(this.props.kudo.receivers, 'name').join(', ')
+    return `Kudos, ${recipients}!`
   }
 
-  postedByActiveUser = () => {
-    let user = this.props.giverId;
-    let poster = this.props.kudo.giver_id;
-    return (user === poster);
+  formattedTimestamp() {
+    const ts = moment(this.props.kudo.given_at)
+    return `At ${ts.format('h:mm a')} on ${ts.format('MMM D, YYYY')}`
   }
 
-  makeEditable = (e) => {
-    this.setState({editing: true});
+  formattedLikeText() {
+    const { likes } = this.props.kudo
+    const numLikes = isEmpty(likes) ? 0 : likes.length
```

Comment:
This was a copy-paste, but you're right! Maybe it was a precaution in case `likes` might be undefined, but it seems that each kudo should have a `likes` attribute defined as an empty array even if there's no likes. Thanks!